packages/metadata/src/inspector.ts

```diff
@@ -234,8 +201,8 @@ export class MetadataInspector {
     methodName = methodName || '';
     const meta: MetadataMap<T[]> =
       options && options.ownMetadataOnly
-        ? Reflector.getOwnMetadata(key, target)
-        : Reflector.getMetadata(key, target);
+        ? Reflector.getOwnMetadata(key.toString(), target)
```

Comment:
Not a blocker but using `.toString()` is messy imo. Wouldn't it be better to do `key.key` or `key.name` with us storing the key as the name property?

Reply:
See my explanation above.